src/cmd/serve.rs

```diff
@@ -763,55 +810,153 @@ pub fn serve(
                                     site = s;
                                 }
                             }
-                            (ChangeKind::Templates, partial_path) => {
-                                let msg = if path.is_dir() {
-                                    format!(
-                                        "-> Directory in `templates` folder changed {}",
-                                        path.display()
-                                    )
-                                } else {
-                                    format!("-> Template changed {}", path.display())
-                                };
-                                console::info(&msg);
-
-                                // A shortcode changed, we need to rebuild everything
-                                if partial_path.starts_with("/templates/shortcodes") {
-                                    if let Some(s) = recreate_site() {
-                                        site = s;
-                                    }
-                                } else {
-                                    println!("Reloading only template");
-                                    // A normal template changed, no need to re-render Markdown.
-                                    reload_templates(&mut site, &path)
-                                }
-                            }
-                            (ChangeKind::StaticFiles, p) => copy_static(&site, &path, &p),
-                            (ChangeKind::Sass, p) => reload_sass(&site, &path, &p),
-                            (ChangeKind::Themes, _) => {
-                                console::info("-> Themes changed.");
-
+                        }
+                        ChangeKind::Templates => {
+                            let partial_paths: Vec<&PathBuf> =
+                                change_group.iter().map(|(p, _, _)| p).collect();
+                            let full_paths: Vec<&PathBuf> =
```

Comment:
when we do get partial and full paths?

Reply:
If I understand your question correctly, we get full paths from the incoming events from `notify`, and we get partial paths from `detect_change_kind`. In the implementation on `next`, check the partial paths to determine whether shortcodes were modified. We pass the full paths to `reload_templates`. I'm not sure if the latter is necessary, so just following what I see on `next`.